tests/resiliency.rs

```diff
@@ -0,0 +1,183 @@
+        toxiproxy_client
+            .all()
+            .unwrap()
+            .iter()
+            .for_each(|(_, proxy)| proxy.delete().unwrap());
+
+        TOXI_ADDR.get_or_init(|| toxi_addr);
+    });
+
+    let local_url = match std::env::var_os("NODE_LOCAL_CACHE") {
+        Some(v) => v.into_string().unwrap(),
+        None => "127.0.0.1:11211".to_string(), // use IPV4 so that it resolves to a single Server
+    };
+    let local_port = PROXY_PORT.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
+
+    let toxi_addr = TOXI_ADDR.get().unwrap();
+    let toxic_local_addr = format!("{}:{}", toxi_addr.ip(), local_port);
+
+    let proxies = vec![ProxyPack::new(
+        format!("local-memcached-{}", local_port),
+        toxic_local_addr.clone(),
+        local_url.clone(),
+    )];
+
+    let toxiproxy_client = ToxiproxyClient::new(toxi_addr);
+    assert!(toxiproxy_client.is_running());
+
+    let proxies = toxiproxy_client.populate(proxies).unwrap();
+    let proxies = proxies
+        .into_iter()
+        .map(|proxy| ProxyDrop { proxy })
+        .collect();
+
+    (proxies, toxic_local_addr)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[ignore = "Relies on a running memcached server and toxiproxy service"]
+    #[test]
+    fn test_set_multi_errors_with_toxiproxy_via_with_down() {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let (proxies, toxic_local_addr) = create_proxies_and_configs();
+
+        let toxic_local_url = "tcp://".to_string() + &toxic_local_addr;
+        let toxic_proxy = &proxies[0];
+
+        let keys = vec!["key1", "key2", "key3"];
+        let values = vec!["value1", "value2", "value3"];
+        let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values).collect();
+
+        let mut clean_client = rt.block_on(async {
+            async_memcached::Client::new("tcp://127.0.0.1:11211".to_string())
+                .await
+                .unwrap()
+        });
+
+        let mut toxic_client =
+            rt.block_on(async { async_memcached::Client::new(toxic_local_url).await.unwrap() });
+
+        let result = rt.block_on(async { clean_client.set_multi(kv.clone(), None, None).await });
+
+        assert!(result.is_ok());
+
+        let _ = toxic_proxy.with_down(|| {
+            rt.block_on(async {
+                let result = toxic_client.set_multi(kv.clone(), None, None).await;
+                assert_eq!(
+                    result,
+                    Err(async_memcached::Error::Io(
+                        std::io::ErrorKind::UnexpectedEof.into()
+                    ))
+                );
+            });
+        });
+    }
+
+    #[ignore = "Relies on a running memcached server and toxiproxy service"]
+    #[test]
+    fn test_set_multi_errors_with_toxiproxy_via_limit_data() {
+        // This test simulates a network error where the response from the server is cut short
```

Comment:
This seems like two seperate tests in one test case? We should probably split them up. (each of the comments)